### PR TITLE
YOLOv1_46686813

### PR DESCRIPTION
**Pull request by Aizhan1202**
This pull request contains project for implementing object-detection on ISIC dataset using YOLOv1 algorithm.
There are four files with code:

- **modules.py** contains all the model blocks, loss function and learning rate scheduler
- **dataset.py** contains functions to load and preprocess the dataset
- **train.py** loads data, trains the model, and saves it to saved_model directory
- **predict.py** contains test script to make example predictions

There are also two .txt files:
- one with the link to **saved model:** the directory was too big for being pushed into github, so the directory with saved file should be downloaded via the link
- second - **target.txt**, which contains target values (bounding box coordinates and class labels)

I was confused with commits. I thought I have to fork the repository in the end, just before I make pull request. So I commited all the changes to my own repository on github.

image

image

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
- Everything committed as one commit, but screenshots provided for some parts of the local commit log.
- Commit has unrelated ipynb files for a UNet implementation too - probably from an earlier solution committed by a different student (?)
- Preprocessing of data seems to be correct.
- No numbers for IoU, mAP or classification accuracy shown. No plots for training/validation loss either.
- Predictions don't seem to be correct. For the shown examples, the bounding boxes and class labels are all same for each image and not predicting the actual location of the lesion.
- Overall performance of the model is not satisfactory. Possible implementation errors.
- Code is okay for normal difficulty

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation (no convergence) -2
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem (result output appears to be the same for examples provide, can't determine if working) -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage (no output numbers) -2
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages, could be more informative -1
Progressive commits used, single commit -2, but done on separate repo (evidence provided), no further action taken, once link posted will receive +2
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time but on separate repo) merged eventually on correct branch, commit history lost though -2 
Feedback required, remove changes to MySolution folder, remove UNet files -2
Request Description good

Please also post link to private repo used to get additional +2 for single commit issue above

**Comment by shakes76** [comment]
You can get up to 4 marks back if you follow instructions in my rubric post.
